emcc.py

```diff
@@ -1082,10 +1086,6 @@ def check(input_file):
 
     if shared.Settings.STRICT:
       shared.Settings.DISABLE_DEPRECATED_FIND_EVENT_TARGET_BEHAVIOR = 1
-    else:
-      # The preprocessor define EMSCRIPTEN is deprecated. Don't pass it to code
-      # in strict mode. Code should use the define __EMSCRIPTEN__ instead.
-      shared.COMPILER_OPTS += ['-DEMSCRIPTEN']
```

Comment:
Why is this location wrong?

Reply:
It was wrong because `shared.COMPILER_OPTS` is read and added to `newargs`. Modifying it after that point is never noticed.